src/cargo-deny/check.rs

```diff
@@ -565,10 +575,49 @@ fn print_diagnostics(
                 Severity::Help => check_stats.helps += 1,
                 Severity::Bug => {}
             }
+
+            // Collect diagnostics for SARIF if needed
+            if let Some(collector) = &sarif_collector
+                && diag.diag.code.is_some()
+            {
+                // Create a diagnostic code based on the check type
+                let diagnostic_code = match pack.check {
```

Comment:
This part...doesn't make sense. It flattens all of the various codes into their category, and assigns blatantly incorrect subcodes without inspecting the actual code.

And then every diagnostic is given the same file and line number, which gives absolutely no information to the end user and again makes every single eg. license diagnostic give the exact same fingerprint, even when it's for a completely different crate, which would seem to me to defeat the point of the fingerprint.